apps/ideas/views.py

```diff
@@ -84,6 +85,11 @@ class IdeaDetailView(AbstractIdeaDetailView):
     )
     permission_required = "a4_candy_ideas.view_idea"
 
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context["page_number"] = get_pagination(self.get_object(), self.module)
```

Comment:
I think instead of using a hard-coded default for how many items there are on a page (`per_page=15`) I would use the value already defined for the `AbstractIdeaListView`, so that if you change it, the new page number function will still compute the correct value. So maybe something like this:
```python
def get_context_data(self, *args, **kwargs):
    context = super().get_context_data(*args, **kwargs)

    context["page_number"] = get_page_number(
        object=self.get_object(),
        objects=self.model.objects.filter(module=self.module),
        paginate_by=AbstractIdeaListView.paginate_by,
    )

    return context
```

Also:
- I would compute the `objects` outside of `get_page_number` to avoid the `obj.__class__.objects`
- I would rename `per_page` to `paginate_by` to be in agreement with the respective parameter of `AbstractIdeaListView`

Reply:
It would be good, but we want to use this function for other modules too, e.g topics.

Reply:
The `AbstractIdeaListView` is inherited by several views, including the `TopicListView`:
```
ProposalListView
SubjectListView
IdeaListView
MapIdeaListView
TopicListView
```

But now that you mention it, shouldn't the new context then be part of the `AbstractIdeaDetailView` instead of `IdeaDetailView`? (so that it also works for topics, map ideas etc. by inheritence?)